pii_check/pii_check_hook.py

```diff
@@ -24,117 +26,108 @@ def get_payload(content, enabled_entity_list, blocked_list):
-            
-        type_pattern_pairs = [blocked_list[i:i + 2] for i in range(0, len(blocked_list), 2)]
+
+        type_pattern_pairs = [blocked_list[i : i + 2] for i in range(0, len(blocked_list), 2)]
         for entity_type, block_pattern in type_pattern_pairs:
-            curr_block_item = {"type": "BLOCK", "entity_type": entity_type, "pattern": block_pattern}
+            curr_block_item = {
+                "type": "BLOCK",
+                "entity_type": entity_type,
+                "pattern": block_pattern,
+            }
             blocked_list_payload.append(curr_block_item)
         payload["entity_detection"]["filter"] = blocked_list_payload
 
     return payload
 
 
-def get_flagged_lines(files):
-    flagged = []
-    for file in files:
-        if os.path.exists(file) and not os.path.isdir(file):
-            with open(file, "r") as fp:
-                lines = fp.readlines()
-                start_flag = False
-                for number, line in enumerate(lines, 1):
-                    if "PII_CHECK:OFF" in line.replace(" ", "").strip() and not start_flag:
-                        start = number
-                        start_flag = True
-                    if "PII_CHECK:ON" in line.replace(" ", "").strip() and start_flag:
-                        end = number
-                        start_flag = False
-                        flagged.append((start, end, file))
-    return flagged
+def get_ignored_lines(filename: str) -> Set[int]:
+    ignored_lines: Set[int] = set()
+
+    with open(filename, "r") as fp:
+        lines = fp.readlines()
+
+    start_flag = False
+    for number, line in enumerate(lines, 1):
+        if "PII_CHECK:OFF" in line.replace(" ", "").strip() and not start_flag:
+            start = number
+            start_flag = True
+        if "PII_CHECK:ON" in line.replace(" ", "").strip() and start_flag:
+            end = number
+            start_flag = False
+            ignored_lines |= set(range(start, end + 1))
+    return ignored_lines
 
 
 def get_response_from_api(content, url, api_key, enabled_entity_list, blocked_list):
     payload = get_payload(content, enabled_entity_list, blocked_list)
     headers = {"Content-Type": "application/json", "X-API-KEY": api_key}
 
-    try:
-        response = requests.post(url, json=payload, headers=headers)
-        response.raise_for_status()
+    response = requests.post(url, json=payload, headers=headers)
+    response.raise_for_status()
+    return response.json()
+
 
-    except HTTPError as e:
-        sys.exit(e)
+def locate_pii_in_file(content: str, file: str, checked: List[str], pii_dict: Dict[str, Any]) -> int:
 
-    pii_result = response.json()
-    return pii_result
+    with open(file, "r") as fp:
+        lines = fp.readlines()
 
+    for line_number, line in enumerate(lines, 1):
+        if content in line:
+            if (pii_dict["location"]["stt_idx"], pii_dict["location"]["end_idx"], line_number) in checked:
+                continue
```

Comment:
Just for my understanding. When this condition will hold true? If I understand correctly, this function will be executed on the 'diff' from a single file at a time.

Reply:
I am not sure what condition you are referring to. Can you please provide the line number? However, in general I've adapted all the functions to run on a single file instead of running on a list of files.

Reply:
[Line 77](https://github.com/privateai/pai-pre-commit-hook/blob/84e8a105deb9e4794e0cac393bb7044fb971e3b5/pii_check/pii_check_hook.py#L77-L78)

More specifically, why are we checking if tuple `(pii_dict["location"]["stt_idx"], pii_dict["location"]["end_idx"], line_number)` is already in `checked` list?

Reply:
Good catch, this is no longer needed. Let me change that.

Reply:
The changes here led me to do a bit more changes. I am using the "hunk" information to be able to locate accurately the detected PII in the text.